pkg/didcomm/dispatcher/outbound.go

```diff
@@ -47,7 +47,7 @@ func NewOutbound(prov provider) *OutboundDispatcher {
 }
 
 // SendToDID sends a message from myDID to the agent who owns theirDID
-func (o *OutboundDispatcher) SendToDID(msg interface{}, myDID, theirDID string) error {
+func (o *OutboundDispatcher) SendToDID(msg service.Headerable, myDID, theirDID string) error {
```

Comment:
The actual usage of msg is https://github.com/hyperledger/aries-framework-go/pull/1026/files#diff-150ad9be6e6f80a3a4239dcaa00d6d5bR78 in the Send function, where it calls marshal on the underlying struct.

(i.e., cares about the annotated struct rather than the accessor methods you created in service.Headerable).

Reply:
Or even more specifically, the method currently cares about being able to marshal the message into JSON. It should also care more that the mandatory fields are present (and from those fields, marshalled).

Reply:
Accessor methods don't affect marshaling. This header was created to easily access common fields. I am going to change `Send` function signature as well.

Reply:
Note: `service.Headerable` is acting the same as `interface{}` in the current usage.

Reply:
You shouldn't only be changing the function signature. The method doesn't actually use anything from the interface.